spot-fix lint errors reported by recent shellcheck

jbertran opened the pull request:
Spotted by @slaperche-scality 

* spot-fix for differing ShellCheck versions between CI and dev laptops
* use a Fedora base image instead of CentOS to get a recent version of ShellCheck

Issue: GH-699

Commit:
spot-fix lint errors reported by recent shellcheck

Issue: GH-699

bert-e commented:
# Hello jbertran,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alxf

* @jbertran

* @ballot-scality

* @carlito-scality

* @nicolast

* @tcarmet

* @zempashi

* @sayf-eddine-scality

* @gdemonet

* @monpote

* @alexandre-allard-scality

* @lucieleonard

* @slaperche-scality

* @thomasdanan

* @ramanduh

* @nootal

* @teddyandrieux

Commit:
switch lint worker to fedora 29

This allows the worker to have an up-to-date ShellCheck version
which CentOS repos do not provide.

Issue: GH-699

Commit:
remove unneeded dep on specific python3 version

We need not restrict to Python 3.6 in test runs. Specifying
Python 3 should be enough for test runs.

Issue: GH-699

jbertran commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue GH-699.

Goodbye jbertran.
